app/models/goods_nomenclatures/nested_set.rb

```diff
@@ -8,6 +8,7 @@ module NestedSet
       one_to_one :tree_node, key: :goods_nomenclature_sid,
                              class_name: 'GoodsNomenclatures::TreeNode',
                              reciprocal: :goods_nomenclature,
+                             graph_use_association_block: true,
```

Comment:
Que pasa?

Reply:
Normally the 'block' is ignored when a relationship is `JOIN`ed - hence why all the time machine stuff gets ignored with `#eager_graph`

This is an option toggle that behaviour which `Sequel` brought in in the January 2023 release. The default behaviour is a bit counter productive for our use case but would likely cause subtle issues if we turned it on everywhere. 

The relationship is new and I'm confident it wont cause problems and solves me needing to set time machine dates when using `association_join`.

Reply:
Thanks Jeremy. Very useful :)